test/common/websocket/codec_test.cc

```diff
@@ -0,0 +1,824 @@
+#include <array>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/websocket/codec.h"
+
+#include "test/common/buffer/utility.h"
+#include "test/test_common/printers.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace WebSocket {
+namespace {
+
+// A single-frame unmasked text message
+// 0x81 0x05 0x48 0x65 0x6c 0x6c 0x6f (contains "Hello")
+//                   H        e        l        l        o
+// 10000001 00000101 01001000 01100101 01101100 01101100 01101111
+// fin - 1
+// opcode - 0x1 - text data
+// mask bit - 0
+// length - 5 bytes 0000101
+TEST(WebSocketCodecTest, decodeSingleUnmaskedFrame) {
+  Buffer::OwnedImpl buffer;
+  Buffer::addSeq(buffer, {0x81, 0x05, 0x48, 0x65, 0x6c, 0x6c, 0x6f});
+
+  Decoder decoder;
+  absl::optional<std::vector<Frame>> frames = decoder.decode(buffer);
+
+  ASSERT_TRUE(frames.has_value());
+  EXPECT_EQ(false, frames.value()[0].masking_key_.has_value());
+  EXPECT_EQ(1, frames.value().size());
+  EXPECT_EQ(5, frames.value()[0].payload_length_);
+
+  std::string text_payload;
+  text_payload.resize(5);
+  (*(frames.value()[0].payload_)).copyOut(0, 5, text_payload.data());
+  EXPECT_EQ("Hello", text_payload);
+}
+
+// A fragmented unmasked text message
+//
+// frame-1: 0x01 0x03 0x48 0x65 0x6c (contains "Hel")
+// 00000001 00000011 H e l
+// fin - 0 (not final frame - either first or not last)
+// text frame - 1 (type info is in initial frame)
+//
+// frame-2: 0x80 0x02 0x6c 0x6f (contains "lo")
+// 10000000 00000010 l o
+// fin - 1 (final frame)
+// continuation frame
+TEST(WebSocketCodecTest, decodeFragmentedUnmaskedTextMessage) {
```

Comment:
From the perspective of this case, this is simply decoding 2 frames, right? Since "Fragments" are not really a concept that the Decoder knows about, I think it would be simpler to call this test "DecodeTwoUnmaskedFrames".